tc/core/tc_executor_new_api.h

```diff
@@ -0,0 +1,125 @@
+ * /// No tensor-related information can be checked so it is the user's
+ * /// responsibility to ensure that shapes and strides match. If the user
+ * /// doesn't then segfault will likely occur.
+ * void uncheckedRun(
+ *     const std::vector<const void*>& inputs,
+ *     const std::vector<void*>& outputs) const;
+ * /// Calls uncheckedRun and profiles the cpu overhead and kernel runtime
+ * /// (microseconds).
+ * /// \returns profiling information (see: tc/core/utils/time.h)
+ * ProfilingInfo profileUnchecked(
+ *     const std::vector<const void*>& inputs,
+ *     const std::vector<void*>& outputs) const;
+ *
+ * As a reminder: for each backend, the specific Backend type lives in
+ *   core/backendname/backendname_backend.h and declares all the required
+ *   dependent **derived** types.
+ * For example:
+ *   CudaBackend is declared in core/cuda/cuda_backend.h
+ *
+ * struct CudaBackend {
+ *   using ExecutorType = CudaTcExecutor;
+ *   using MappingOptionsType = CudaMappingOptions;
+ *   using CompilationResultType = CudaCompilationResult;
+ *   using RTCFunctionType = CudaRTCFunction;
+ * };
+ *
+ * The correspondence is 1 TcExecutor for 1 compiled
+ *   tuple<TC function, input shapes, Backend::MappingOptions>.
+ */
+template <typename Backend>
+class TcExecutor {
+ protected:
+  TcExecutor(
+      const std::vector<TensorInfo>& inputsInfo,
+      const std::vector<TensorInfo>& outputsInfo,
+      const tc2halide::HalideComponents& halideComponents,
+      const typename Backend::CompilationResultType& compilationResult);
+
+ public:
+  TcExecutor(TcExecutor&&) = delete;
+  TcExecutor& operator=(TcExecutor&&) = delete;
+  TcExecutor(const TcExecutor&) = delete;
+  TcExecutor& operator=(const TcExecutor&) = delete;
+
+  /// Run can be called multiple times given a compilation, inputs are allowed
+  /// to change in that their data pointer is allowed to change.
+  /// Sizes and strides must remain constant otherwise this is an error.
+  /// The only thing that is allowed to change across runs is the input
+  /// and output pointers base address.
+  /// It is the caller's responsibility to ensure proper non-aliasing (or
+  /// advanced aliasing) properties of the input and output tensors.
+  void run(
+      const std::vector<const DLConstTensor*>& inputs,
+      const std::vector<const DLTensor*>& outputs) const;
+
+  /// Calls run and profiles the cpu overhead and kernel runtime (microseconds).
+  /// \returns profiling information (see: tc/core/utils/time.h)
+  ProfilingInfo profile(
+      const std::vector<const DLConstTensor*>& inputs,
+      const std::vector<const DLTensor*>& outputs) const;
+
+  /// It may be necessary to clear the RTC manually because it can throw and
+  /// we can't have that in the RTC destructor.
+  void clearRuntimeCompiledFunction();
+
+ public:
+  // For inspection purposes
+  const std::string compiledSource;
+
+ protected:
+  /// Used to check proper metadata when calling run
```

Comment:
Doxygen has groups (`@{` / `@}` or `\{` / `\}`) that make the comment apply to multiple fields, like in this case, rather than to the first one.